### ISSUE TITLE
JSDoc shown types 'any' always

### ISSUE THREAD

**Issue by aliuq** [microsoft/vscode#146344]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.65.2
- OS Version: Windows 11

Steps to Reproduce:

Launch with `code --disable-extensions`

`@params` type only show `any`

![image](https://user-images.githubusercontent.com/11707934/160910403-704f9883-248c-44f6-8580-7f059de94f9f.png)

**Comment by IllusionMH** [comment]
/needsMoreInfo

Is it reproducible in latest VS Code 1.66.0?

P.S. You have better chance to receive help if you post actual code in addition to screenshots. Nobody likes to retype code from screenshots to reproduce your issue

**Comment by mjbvz** [comment]
Also please always share a self contained code example as text

**Comment by aliuq** [comment]
> /needsMoreInfo
> 
> Is it reproducible in latest VS Code 1.66.0?
> 
> P.S. You have better chance to receive help if you post actual code in addition to screenshots. Nobody likes to retype code from screenshots to reproduce your issue

Sorry no code added before, now I added the code from [here](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns), and same issue in v1.66.0

![image](https://user-images.githubusercontent.com/11707934/160965532-83bf711d-3aec-403c-95bf-a88ad08dc2a4.png)

**Comment by aliuq** [comment]
> Also please always share a self contained code example as text

Thanks for your advice, I will remember it

```ts
// Parameters may be declared in a variety of syntactic forms
/**
 * @param {string}  p1 - A string param.
 * @param {string=} p2 - An optional param (Google Closure syntax)
 * @param {string} [p3] - Another optional param (JSDoc syntax).
 * @param {string} [p4="test"] - An optional param with a default value
 * @returns {string} This is the result
 */
function stringsStringStrings(p1, p2, p3, p4) {
  // TODO
}
```

**Comment by yume-chan** [comment]
Types in JSDoc only works in JavaScript files, not TypeScript files.

**Comment by aliuq** [comment]
> Types in JSDoc only works in JavaScript files, not TypeScript files.

Thank you. I think the `@param` type of JSDoc can be used in typescript

**Comment by yume-chan** [comment]
You can use `@param` and `@return` for descriptions, but not types. TypeScript type notations must be used.

**Comment by mjbvz** [comment]
Good catch @yume-chan! Yes JSDoc types are only observed in JS files. Closing this as by-design
